core/src/main/scala/org/http4s/multipart/MultipartParser.scala

```diff
@@ -776,15 +753,28 @@ object MultipartParser {
         racc: Stream[F, Byte],
         limitCTR: Int): SplitFileStream[F] =
       if (limitCTR >= maxBeforeWrite)
-        Pull
-          .eval(F.delay(Files.createTempFile("", "")))
-          .flatMap { path =>
-            (for {
-              _ <- Pull.eval(lacc.through(writeAll[F](path, blocker)).compile.drain)
-              split <- streamAndWrite(s, state, Stream.empty, racc, 0, path)
-            } yield split)
-              .handleErrorWith(e => Pull.eval(cleanupFile(path)) >> Pull.raiseError[F](e))
-          }
+        ???
```

Comment:
My confusion is as thus: it seems a temp file is created but not always cleaned up. That does not sound right to me, so I tried to read through it to get a complete picture but I have to admit - I don't quite understand what's supposed to be happening here so I am left a little confused. Any help would be greatly appreciated.

Reply:
Actually, I might be able to work it through. I'll give it a shot now.